### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Czupryn013**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# Byte-compiled / optimized / DLL files
```
Comment (mstopa) [inline .gitignore:1]: Please make a separate directory for app, and another one for training notebooks

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+__pycache__/
```
Comment (mstopa) [inline .gitignore:2]: `setup.py` or `setup.cfg project.toml`

File: svm.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# Create a classifier: a support vector classifier
+clf = svm.SVC(gamma=0.001)
+
+# Split data into 50% train and 50% test subsets
+X_train, X_test, y_train, y_test = train_test_split(
+    data, digits.target, test_size=0.5, shuffle=False
+)
+
+# Learn the digits on the train subset
+clf.fit(X_train, y_train)
```
Comment (mstopa) [inline svm.py]: please add saving the model too

File: svm_controller.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from flask import Flask, request
+import pickle
+import numpy as np
+
+app = Flask(__name__)
+
+model = pickle.load(open("svv.sav", 'rb'))
```
Comment (mstopa) [inline svm_controller.py]: please consider providing model path through default function kwarg or argparse

File: svm_controller.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+@app.route("/digits", methods = ["GET"])
```
Comment (mstopa) [inline svm_controller.py]: maybe you could create a healthcheck endpoint like https://pypi.org/project/py-healthcheck/ ?

File: svm_controller.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    return "This is a digits recogniton app.", 200
+
+@app.route("/digits", methods = ["POST"])
+def get_digit():
+    request_data = request.get_json()
+    digit = request_data.get("digit")
+    if not digit or len(digit) != 64 or type(digit) != list: return "Incorrect json body", 400
+
+    for element in digit:
+        if not isinstance(element, int) or not all(ch in ['0', '1'] for ch in bin(element)[2:]):
```
Comment (mstopa) [inline svm_controller.py]: `all(ch in ['0', '1'] for ch in bin(element)[2:])` will always be true for integers

**Comment by mstopa** [comment]
Notebooks are understandable to read, it is ok to keep them in their current form

File: svm_controller.py (hunk 6 above)
Comment (Czupryn013) [reply svm_controller.py]: Fixed

File: .gitignore (hunk 1 above)
Comment (Czupryn013) [reply .gitignore:1]: Done

File: svm.py (hunk 3 above)
Comment (Czupryn013) [reply svm.py]: Done

File: .gitignore (hunk 2 above)
Comment (Czupryn013) [reply .gitignore:2]: Done

File: training/svm.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    f"Classification report for classifier {clf}:\n"
+    f"{metrics.classification_report(y_test, predicted)}\n"
+)
+
+con_matrix = confusion_matrix(y_test, predicted)
+disp = ConfusionMatrixDisplay(confusion_matrix=con_matrix)
+disp.plot()
+
+# jupyter-kernelspec uninstall .venv
+# ipython kernel install --name=.venv
```
Comment (mstopa) [inline training/svm.py]: del?

File: training/svm.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
+from sklearn import datasets, svm, metrics
+import numpy as np
+from tensorflow.keras.datasets import mnist
+from sklearn.model_selection import train_test_split
+import matplotlib.pyplot as plt
+import pickle
+from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay
```
Comment (mstopa) [inline training/svm.py]: isort?

File: app/desktop_train.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+GRID_COLOR = (211, 211, 211)
+
+conf = ""
+drawing_array = np.zeros((28, 28))
+current_num = 0
+# running first time or resteting use 21 and 22 line, otherwise 19 and 20
+count = np.load("count.npy")
+test_data = np.load("test_data.npy")
+# count = np.array([999])
+# test_data = np.zeros((1000, 784))
```
Comment (mstopa) [inline app/desktop_train.py]: add condition instead of instructions in the comment

Comment (Czupryn013) [reply app/desktop_train.py]: Done

File: training/svm.py (hunk 7 above)
Comment (Czupryn013) [reply training/svm.py]: Done

**Comment by mstopa** [comment]
@Czupryn013 please check if the app works and if so, please merge this pull request

**Pull request merged**